### ISSUE TITLE
ps2 keyboard not reliable

### ISSUE THREAD

**Issue by fanoush** [Basman74/Next186#1]
I have two different USB keyboards that can do PS/2 too, I also tried them with three different usb micro OTG adapters (micro male to full usb female) but getting keyboard working after power is hit and miss. Even if I see BIOS/FreeDOS to say ps2 keyboard detected sometimes/often it does not work after power on. Powering off and on (not touching board or connector at all) make it working in ~50% of cases.

**Comment by Basman74** [comment]
Hello,
I have two composite USB keyboards (ie can speak either PS/2 or USB) that work reliably here PROVIDED I have an adequate and clean USB supply for the Ohm (Next186 makes the Ohm board draw at least 150mA from +5V when running!) and Next186 synthesis is within the timing constraints of the intended target board...
What are you running Next186/Ohm from? USB wall-wart? PC USB host port? Thanks.
Regards,Valentin 

    On Tuesday, 6 February 2018, 19:57, fanoush <notifications@github.com> wrote:
 

 I have two different USB keyboards that can do PS/2 too, I also tried them with three different usb micro OTG adapters (micro male to full usb female) but getting keyboard working after power is hit and miss. Even if I see BIOS/FreeDOS to say ps2 keyboard detected sometimes/often it does not work after power on. Powering off and on (not touching board or connector at all) make it working in ~50% of cases.—

**Comment by fanoush** [comment]
It is 2A usb power adapter (Nook HD tablet charger)  I use with raspberry pi without issues. Power should be enough but maybe it is noisy. I'll try different one and different power cable just to be sure. Thanks for the tip.

After poweron both keyboards behave same and blink twice with numlock leds and then either keypresses work or not. And BTW I use left usb, the one next to power, the other one next to hdmi is perhaps hardcoded to mouse(?) as it does not work with keyboard.

**Comment by Basman74** [comment]
That is correct. Also yes, having a good quality USB cable helps A LOT for reliable operation I've found. Also yes, USB1 is hardwired for PS/2 mouse while USB2 is for the keyboard. I will elaborate on that in the setup docs now :) Cheers.

**Comment by fanoush** [comment]
Just to let you know that changing power adapter or cable did not help. This time I used 3A one  for raspberry pi https://www.wdc.com/en-ie/products/wdlabs/wd-pidrive-power-kit-3a.html

So I guess it is those keyboards.  Perhaps it is some timing issue somewhere? Any idea abut some timeout value or something that I could change?

**Comment by Basman74** [comment]
I have confirmed the issue you've raised in this thread and brought it to the Next186 author's attention.

**Comment by Basman74** [comment]
As per the discussion from our last thread, it appears my attempt to add HDMI audio to Next186 was not successful :(  Indeed, it appears to add (for as yet unknown reasons, at least to me) unwanted instability to the PS/2 keyboard controller but there are also other issues with it as well. I have since withdrawn the suspect firmware build from the repository. For future reference I suggest using Nicolae's original HDMI + GPIO audio firmware example and sources instead. Regards, Valentin

**Comment by fanoush** [comment]
Well maybe if you marked the firmware with alpha or beta in name it could be useful to leave it there in the repository. In that case however it would be also beneficial to upload also matching source so someone could look at it and possibly help to fix it.

So far the only issue I see is the keyboard detection at power on. When it is detected it seems to work. So far it did not stop working until next power cycle and also it did not miss key presses or produce wrong output. So overall it looks useful even in current state.
